### ISSUE TITLE
Enum properties cannot be decoded

### ISSUE THREAD

**Issue by woppa684** [LIonWeb-org/lioncore-typescript#57]
Only builtins are decoded, enum properties throw an error.

**Comment by dslmeinte** [comment]
Could you maybe provide a failing unit test?

**Comment by dslmeinte** [comment]
I found the problem: https://github.com/LIonWeb-org/lioncore-typescript/blob/main/src/deserializer.ts#L96
Just haven't fixed it yet ;) A (now still-)failing unit test would nevertheless be helpful because this happens on M1-level which has more degrees of freedom.

**Comment by woppa684** [comment]
I made the exisiting unit tests fail in the PR above. Is this the preferred way of contributing?

**Comment by dslmeinte** [comment]
I don't know about any _preferred_ way of contributing, but it's certainly a fine one: thanks! :)

I'm thinking about how/when to merge this: stay tuned.

**Comment by dslmeinte** [comment]
I suspect that the serialization of enums also already didn't work.

**Comment by dslmeinte** [comment]
(And neither did the Ecore import w.r.t. enums.)

**Comment by dslmeinte** [comment]
Found a more fundamental problem: the runtime/in-memory representation of enum literals is, in general, entirely free, so this needs an explicit 2-way conversion for (de-)serialization.

**Comment by woppa684** [comment]
Yeah, already thought so indeed. You are free to have a string-enum, number-enum, const enum or whatever so there is no single way of deserializing to an enum from the client-domain. For deserialization it would probably need something like: this in the `ModelAPI`?

```ts
enumLiteralFor: (literal: EnumerationLiteral) => any
```

**Comment by dslmeinte** [comment]
Exactly, and a symmetric one for serialization.

**Comment by dslmeinte** [comment]
Should be fixed by PR #60 : could you also check/review, @woppa684 ?
